bin/csv_from_oai.rb

```diff
@@ -0,0 +1,80 @@
+require 'csv'
+require 'oai'
+require 'byebug'
+
+# Input for dynamic email
+puts "Enter your email address:"
+email = gets.chomp
+
+# OAI client setup
+client = OAI::Client.new(
+  "http://oai.adventistdigitallibrary.org/OAI-script",
+  headers: { from: email },
+  parser: 'libxml'
+)
+
+opts = {
+  metadata_prefix: "oai_adl"
+}
+
+CSV.open('csv_from_oai.csv', 'wb') do |csv|
+  # Write the headers to the CSV file
+  csv << ['set', 'identifier', 'derivative_type', 'URL']
+
+  sets = ["adl:thesis",
+    "adl:periodical",
+    "adl:issue",
+    "adl:article",
+    "adl:image",
+    "adl:book",
+    "adl:other"
+  ]
+
+  sets.each do |set|
+    records = client.list_records(opts.merge(set: set))
+    records.each do |r|
```

Comment:
Reading Notes: the current implementation, which works great for testing, will only get the first page of each set.

This should have `records.full.each do |record|`

I'm wondering, how might we structure our code to allow us to choose if we want to run the `records.full.each` or the `records.each`?  For most tests, we probably don't want to run `records.full.each` but when we generate the actual CSV, we'll want that.

Reply:
Updated